lib/WebService/Async/CustomerIO/Customer.pm

```diff
@@ -0,0 +1,216 @@
+package WebService::Async::CustomerIO::Customer;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::Async::CustomerIO::Customer - Class for working with customer.
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+=cut
+
+use Carp qw();
+
+
+=head2 new(%params) -> obj
+
+Creates customer object. This method just creates an object, to sent this data to api, after creation should be called upsert method.
+
+parameters:
+- id: the unique identifier for the customer.
+- email: optional. the email address of the user.
+- created_at: optional. the unix timestamp from when the user was created in your system
+- attributes: hashref which contains custom attributes to define the customer.
+
+=cut
+
+sub new {
+    my ($cls, %param) = @_;
+
+    $param{$_} or Carp::croak "Missing required argument: $_" for (qw(id api_client));
+
+    my $self = +{map { $_ => $param{$_} } qw(id email created_at attributes api_client)};
```

Comment:
I don' think this is needed?

Just use the existing hash:

```
return bless \%param, $class;
```

Reply:
You're right there's no point to copy it twice. Fixed